trie.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from piece import Piece
+import time
+
+class Trie:
+	def __init__(self):
+		self.root = TrieNode("root", None)
+
+	def get_root(self):
+		return self.root
+
+	def add_node(self, points, value):
+		p = 0
+		node = self.root
+		while points[p] in node.get_children():
+			node = node.get_children()[points[p]]
+			p += 1
+			if p == len(points):
+				node.set_value(value)
+				return node
+		while p < len(points) - 1:
+			node = node.add_child(None, points[p])
+			p += 1
+		node.add_child(value, points[p])
+		return node
+
+
+
+class TrieNode:
+	def __init__(self, value, point):
+		self.value = value
+		self.point = point
+		self.children = {}
+
+	def get_value(self):
+		return self.value
+
+	def set_value(self, value):
+		self.value = value
+
+	def get_point(self):
+		return self.point
+
+	def get_children(self):
+		return self.children
+
+	def add_child(self, value, point):
+		self.children[point] = TrieNode(value, point)
+		return self.children[point]
+
+
+
+def get_points_from_shape(shape, start):

```

Comment:
Do you mind explaining the logic of including the code below this line in this file. 
- BFS/DFS I get, but I think they could be part of the trie class.
- The other classless functions seem like they should probably be moved elsewhere if they're not part of the trie implementation.
- It seems like the raw code was just used for smoke testing? If that's the case it shouldn't be commited, otherwise I still think it should be moved elsewhere.

Reply:
Removed testing code and moved the other functions into a "TrieBuilder" class